tests/acceptance/01_vars/02_functions/validjson_trailing_bogus_data.cf

```diff
@@ -0,0 +1,43 @@
+##############################################################################
+#
+# Test validjson() evaluates to !any:: when there is trailing bogus data after
+# termination for JSON object.
+#
+##############################################################################
+
+body common control
+{
+  bundlesequence => { "test", "check" };
+  version => "1.0";
+}
+
+##############################################################################
+
+bundle agent test
+{
+  meta:
+      "description" -> { "CFE-4080" }
+        string => "Test validjson() with trailing bogus data";
+}
+
+##############################################################################
+
+bundle agent check
+{
+  classes:
+      "ok"
+        and => {
+          not(validjson('"some": [ "json" ] }')),
+          not(validjson('{ "some": [ "json" ] } [')),
+          not(validjson('["some", "json"]!')),
+          not(validjson('    ["some", "json"]a')),
+          not(validjson('["some", "json"] {"foo": "var"}   ')),
+          validjson('{"test": [1, 2, 3]}'),
```

Comment:
My comment in the libntech PR could apply here as well:

https://github.com/cfengine/libntech/pull/178